internal/ovirt/resource_ovirt_vm.go

```diff
@@ -219,6 +218,11 @@ func (p *provider) vmCreate(
 		diags = f(data, params, diags)
 	}
 
+	diags = handleVMPlacementPolicy(client, data, params, diags)
+	if diags.HasError() {
```

Comment:
Previously, `diags` was only used to append errors in the handle functions. I am not sure, if checking for an error and returning the `diags` is desired behavior. If not, I can easily remove it again.

Reply:
If diags is not empty it should be returned instead of trying to create the VM.